firebase-firestore/src/main/java/com/google/firebase/firestore/local/LruGarbageCollector.java

```diff
@@ -15,16 +15,82 @@
 package com.google.firebase.firestore.local;
 
 import android.util.SparseArray;
+import com.google.firebase.firestore.FirebaseFirestoreSettings;
 import com.google.firebase.firestore.core.ListenSequence;
+import com.google.firebase.firestore.util.Logger;
 import java.util.Comparator;
 import java.util.PriorityQueue;
 
 /** Implements the steps for LRU garbage collection. */
-class LruGarbageCollector {
+public class LruGarbageCollector {
+  public static class Params {
```

Comment:
Would it be possible to follow the Builder pattern here or add the static factory methods to `LruGarbageCollector` itself?

Reply:
This still feels very un-Java esque to me. The C++/TS approach might be something that just doesn't port well.

An easy way to fix this might be to return an `LruGarbageCollector` from `Params.Default()` and so forth. What do you think?

Reply:
I would still rather not do this. The point is to separate data and behavior. We have encapsulation of the data in a simple struct, and behavior that reads that struct for its parameters. I also don't think it's particularly unfamiliar for Java. Having an `Options` or `Params` object is not unheard of in the constructor to a class, or as an argument to a method. In fact, a search for the exact classname, `static class Params` turns up a number of results using exactly this pattern.